### PR TITLE
CSS: Don’t trim whitespace of undefined custom property

### PR DESCRIPTION
**Pull request by andersk**
### Summary ###

Fixes gh-5105

### Checklist ###

* [x] New tests have been added to show the fix or feature works
* [x] Grunt build and unit tests pass locally with these changes
* [ ] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

**Comment by mgol** [comment]
Thanks for the PR with a test!

We’ll discuss what approach we want to take here and we’ll get back to you.

File: test/unit/css.js [hunk 1]
```diff
@@ -1802,6 +1802,7 @@ QUnit.testUnlessIE( "css(--customProperty)", function( assert ) {
 	assert.equal( $elem.css( "--prop10" ), "val10", "Multiple preceding and following escaped unicode whitespace trimmed" );
 	assert.equal( $elem.css( "--prop11" ), "val11", "Multiple preceding and following unicode whitespace trimmed" );
 	assert.equal( $elem.css( "--prop12" ), "\u000Bval12\u000B", "Multiple preceding and following non-CSS whitespace reserved" );
+	assert.equal( $elem.css( "--nonexistent" ), undefined );
```
Comment (mgol) [inline test/unit/css.js:1805]: Please also add a test for a CSS variable that is defined but with no meaningful value: e.g. `--empty: ;`. This should also return `undefined`. I wish we were able to return an empty string there but I don't see a way so let's at least test the current behavior.

**Review by mgol — CHANGES_REQUESTED** [review]
Just one comment, otherwise it looks good to me. Thanks!

File: test/unit/css.js (hunk 1 above)
Comment (andersk) [reply test/unit/css.js:1807]: Added.

**Review by mgol — CHANGES_REQUESTED** [review]
One of the tests failed in Chrome; can you have a look?

Disregard the `selector` `:has` failures, they are caused by #5098 and will be fixed when #5107 lands.

**Comment by andersk** [comment]
This is a cross-browser difference with `--empty: ;`. In Firefox, `.getPropertyValue("--empty")` returns the empty string `""`, while in Chrome, it returns a space `" "`.

I’ve changed the test to `--empty:;` without the space to avoid this issue. We can consider further normalizing this cross-browser difference, but since that’s neither a crash nor a regression, I think we should consider it separately.

**Review by mgol — APPROVED** [review]
Makes sense, thanks! I'll need to ask you to rebase the PR once #5107 lands so that the build is green here. Other than that, LGTM.

**Comment by andersk** [comment]
Rebased.

**Comment by andersk** [comment]
Backported to `3.x-stable` as #5118.

**Comment by mgol** [comment]
Landed on `main` in ed306c0261ab63746040e5d58bb4477c3069a427 and on `3.x-stable` in c0db6d70320ad6fb29a6a89aae3811a81b5adedf. Thanks!

**Comment by mgol** [comment]
> I’ve changed the test to `--empty:;` without the space to avoid this issue. We can consider further normalizing this cross-browser difference, but since that’s neither a crash nor a regression, I think we should consider it separately.

I submitted #5120 to handle this case.

**Pull request merged**